src/compiler/scala/tools/nsc/typechecker/Macros.scala

```diff
@@ -624,16 +624,17 @@ trait Macros extends MacroRuntimes with Traces with Helpers {
             macroLogVerbose(s"cancelled macro expansion because of $reason: $expandee")
             onFailure(typer.infer.setError(expandee))
           } else try {
+            // also see http://groups.google.com/group/scala-internals/browse_thread/thread/492560d941b315cc
+            val expandeeDuplicate = duplicateAndKeepPositions(expandee)
             val expanded = {
               val runtime = macroRuntime(expandee)
               if (runtime != null) macroExpandWithRuntime(typer, expandee, runtime)
               else macroExpandWithoutRuntime(typer, expandee)
             }
             expanded match {
               case Success(expanded) =>
-                // also see http://groups.google.com/group/scala-internals/browse_thread/thread/492560d941b315cc
-                val expanded1 = try onSuccess(duplicateAndKeepPositions(expanded)) finally popMacroContext()
```

Comment:
Seems removing this `duplicate` breaks the `macro-auto-duplicate` test. Its macro generates

```
{
  val x = 4;
  x
}.$plus({
  val x = "2";
  x
}): String
```

which fails with

```
 found   : Int
 required: String
  println(Macros.foo)
                 ^
```

Reply:
Also macro-duplicate is broken:

```
partest --update-check \
  /home/travis/build/scala/scala/test/files/run/macro-auto-duplicate \
  /home/travis/build/scala/scala/test/files/run/macro-duplicate
```